src/python/pants/backend/python/rules/coverage_integration_test.py

```diff
@@ -143,42 +144,84 @@ def test_coverage(self) -> None:
+                {tmpdir_relative}/src/python/project/__init__.py                        0      0      0      0   100%
                 {tmpdir_relative}/src/python/project/lib.py                             6      0      0      0   100%
+                {tmpdir_relative}/src/python/project/lib_test.py                        3      0      0      0   100%
+                {tmpdir_relative}/src/python/project/random.py                          2      2      0      0     0%
                 {tmpdir_relative}/tests/python/project_test/no_src/test_no_src.py       2      0      0      0   100%
+                {tmpdir_relative}/tests/python/project_test/test_arithmetic.py          3      0      0      0   100%
+                {tmpdir_relative}/tests/python/project_test/test_multiply.py            3      0      0      0   100%
                 -----------------------------------------------------------------------------------------------
-                TOTAL                                                             8      0      0      0   100%
+                TOTAL                                                            19      2      0      0    89%
                 """
             )
             in result.stderr_data
         )
+
+    def _assert_html_report(self, result: PantsResult, build_root: str) -> None:
+        assert "Wrote html coverage report to `dist/coverage/python`" in result.stderr_data
+        html_cov_dir = Path(build_root, "dist", "coverage", "python", "htmlcov")
+        assert html_cov_dir.exists() is True
+        assert (html_cov_dir / "index.html").exists() is True
+
+    def _assert_xml_report(self, result: PantsResult, build_root: str) -> None:
+        assert "Wrote xml coverage report to `dist/coverage/python`" in result.stderr_data
+        xml_coverage = Path(build_root) / "dist" / "coverage" / "python" / "coverage.xml"
+        assert xml_coverage.exists() is True
+        root = ElementTree.parse(xml_coverage.as_posix()).getroot()
+        assert root.tag == "coverage"
+        assert "version" in root.attrib
+        assert len(root) == 2
+        assert root[0].tag == "sources"
+        assert root[1].tag == "packages"
+
+    def test_coverage_xml(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=xml")
+            self._assert_xml_report(result, build_root)
+
+    def test_coverage_html(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=html")
+            self._assert_html_report(result, build_root)
```

Comment:
Thanks for deleting the other test. I still don't think we get much value from having these when we already have `test_coverage_html_and_xml`. That test will already validate that we correctly generate HTML + XML.

That test also validates that it's safe to do multiple reports, which is something these tests do as well because they generate both `console` and `html/xml`.